src/renderer/containers/Header/UserMenu.tsx

```diff
@@ -88,25 +108,38 @@ export const UserMenu: React.FC<{
           Logout
         </MenuItem>
       </Menu>
-      <Dialog
-        fullScreen={fullScreen}
+      <ConfirmationModal
+        title="Change your Display Name"
+        open={openNameChangePrompt}
+        onClose={handleClose}
+        onSubmit={onChangeName}
+        fullWidth={false}
+      >
+        <Typography paragraph={true}>Enter the new name in the field below</Typography>
+        <TextField
+          required={true}
+          id="displayName"
+          variant="outlined"
+          autoFocus={true}
+          key={`displayName`} // Change key on new mode to force field to reset
+          name="displayName"
+          value={nameChange || name}
```

Comment:
> `nameChange || name`
Doing this is not good. When you clear the entire input e.g. with Ctrl+backspace, it defaults to `name`. It makes for a really awkward typing experience. I would recommend creating a new component `NameChangeDialog` which has a local state for the input box value. You can also add a use effect hook to reset it to `user.displayName` before opening.